apis/composition.yaml

```diff
@@ -0,0 +1,110 @@
+              imageAutomationController:
+                create: false
+              imageReflectionController:
+                create: false
+      patches:
+        - type: PatchSet
+          patchSetName: Common
+        - fromFieldPath: spec.parameters.operators.flux.version
+          toFieldPath: spec.forProvider.chart.version
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: spec.parameters.operators.flux.version
+            strategy: string
+            string:
+              fmt: https://github.com/fluxcd-community/helm-charts/releases/download/flux2-%[1]s/flux2-%[1]s.tgz
+          toFieldPath: spec.forProvider.chart.url
+
+    - name: syncFlux
+      base:
+        apiVersion: helm.crossplane.io/v1beta1
+        kind: Release
+        spec:
+          forProvider:
+            namespace: flux-system
+            chart:
+              name: flux2-sync
+              repository: https://fluxcd-community.github.io/helm-charts
+              # default version
+              version: "1.7.2"
+            values:
+              gitRepository:
+                spec:
+                  timeout: 20s
+                  gitImplementation: go-git
+              kustomization:
+                spec:
+                  sourceRef:
+                    kind: GitRepository
+      patches:
+        - type: PatchSet
+          patchSetName: Common
+        - fromFieldPath: spec.parameters.operators.flux-sync.version
+          toFieldPath: spec.forProvider.chart.version
+        - fromFieldPath: spec.parameters.id
+          toFieldPath: metadata.annotations[crossplane.io/external-name]
+        - fromFieldPath: spec.parameters.source.git.interval
+          toFieldPath: spec.forProvider.values.gitRepository.spec.interval
+        - fromFieldPath: spec.parameters.source.git.interval
+          toFieldPath: spec.forProvider.values.kustomization.spec.interval
+        - fromFieldPath: spec.parameters.source.git.ref.name
+          toFieldPath: spec.forProvider.values.gitRepository.spec.ref.name
+        - fromFieldPath: spec.parameters.source.git.url
+          toFieldPath: spec.forProvider.values.gitRepository.spec.url
+        - fromFieldPath: spec.parameters.source.git.path
+          toFieldPath: spec.forProvider.values.kustomization.spec.path
+        - fromFieldPath: spec.parameters.kubeConfigSecretRef.name
+          toFieldPath: spec.forProvider.values.kustomization.spec.kubeConfig.secretRef.name
+        - fromFieldPath: spec.parameters.kubeConfigSecretRef.key
+          toFieldPath: spec.forProvider.values.kustomization.spec.kubeConfig.secretRef.key
+        - fromFieldPath: spec.parameters.id
+          toFieldPath: spec.forProvider.values.gitRepository.spec.ref.branch
+          transforms:
+            - type: match
+              match:
+                patterns:
+                  - type: regexp
+                    regexp: '.*'
+                    result: ''
```

Comment:
why are we doing it?

Reply:
because the sync helm-chart defines ref.branch master per default in values - we want to use  ref.name because server-side-apply is an issue when using ref.tag or ref.branch switch - so its better to to use ref.name

Reply:
can you leave an inline comment? It's totally unobvious when you look at the code

Reply:
done